Sources/MessagingInApp/Gist/Managers/Models/Message.swift

```diff
@@ -164,7 +164,16 @@ extension Array where Element == Message {
             switch ($0.priority, $1.priority) {
             case (let priority0?, let priority1?):
                 // Both messages have a priority, so we compare them.
-                return priority0 < priority1
+
+                // if the messages do not have the same priority, sort by the priority.
+                if priority0 != priority1 {
+                    return priority0 < priority1
+                }
+
+                // If messages have same priority, sort by something else to assert that the function return value is always the same no matter if the order of the input array is different.
+
+                // Because the priorities are the same, it doesn't matter what message is next. Use a unique value to perform the comparison.
+                return $0.instanceId < $1.instanceId
```

Comment:
I was encountering a bug during sample app testing that this code fixes. 

Reproduce steps:
* Send 2+ messages with same element id and priority to device
* Open app and wait until 1 inline View is in foreground and View is showing a message. 
* Wait 10-60 seconds. You should notice that the inline View will randomly change to displaying the other message that you sent to the device, without you touching the screen. 

This bug fix makes it so that when `sortByMessagePriority` is called, as output will always be the same for any given input. [See added tests to learn more](https://github.com/customerio/customerio-ios/pull/756/files#diff-d167fc3626f2b47618e1db865ee4d5631fd993ba64fd17a303fe77c1078f9d0fR68)

Reply:
FYI, I have a follow-up PR that modifies this logic. I found a bug in it. https://github.com/customerio/customerio-ios/pull/760